app/controllers/register_controller.rb

```diff
@@ -0,0 +1,29 @@
+class RegisterController < ApplicationController
+
+  def new
+  end
+
+  def create
+    @user = User.new(user_params)
+    if User.exists?(email: user_params[:email])
+      flash[:error] = "Email already in use. Try a different email."
+      render :new
+    elsif params[:confirm_pass] != params[:password]
+      flash[:error] = "Unable to create account: Passwords don't match."
+      render :new
+    elsif params[:confirm_pass] == params[:password] && @user.save
+      flash[:notice] = "Your account has been created."
+      redirect_to "/profile"
+    else
+      flash[:error] = "Unable to create account: Required information missing."
+      redirect_to request.referer
```

Comment:
Let's look at possibly going back to render for these, just in the name of consistency